### PR TITLE
Harden OpenCode usage analytics confidence and persistence

### LINKED ISSUE #803: Harden OpenCode token analytics accuracy and confidence
## Problem
OpenCode token analytics exists, but aggregate/historical totals are not yet reliable enough to treat as canonical across CAR surfaces.

Current risks:
- No single canonical usage contract across all OpenCode paths.
- Historical parsing is heuristic (`.opencode/sessions/**/*.json`) and vulnerable to schema drift.
- Delta vs cumulative semantics are ambiguous.
- Missing provider-specific parse/fallback telemetry and targeted tests.

## Goal
Make OpenCode token usage accuracy reliable enough for production analytics decisions.

## Scope
1. Establish canonical OpenCode usage contract
- Add first-class normalized usage field(s) to OpenCode run results/events.
- Ensure exactly one normalized usage snapshot per logical turn.

2. Persist normalized usage at ingestion time
- Persist normalized turn usage into CAR-owned state under `.codex-autorunner/` (or existing canonical internal store), rather than relying only on reparsing `.opencode/sessions` artifacts.

3. Harden aggregation logic
- Define and enforce delta vs cumulative semantics.
- Add guards for malformed timestamps/payloads to skip bad records without aborting full aggregation.
- Unify repo attribution behavior across Codex/OpenCode paths.

4. Confidence and observability
- Add provider/source confidence metadata (e.g. `live_stream`, `persisted_normalized`, `estimated`, `billing_reconciled`) in usage API surface.
- Add parse/fallback counters and timing metrics for OpenCode usage ingestion/aggregation.

5. Tests
- Add focused tests for OpenCode usage ingestion + aggregation:
  - payload shape variants
  - cumulative vs delta behavior
  - malformed timestamp/payload handling
  - repo attribution edge cases
  - cache merge behavior with OpenCode + Codex

## Non-goals
- Removing usage endpoints.
- Billing reconciliation implementation (can be follow-up).

## Acceptance criteria
- OpenCode usage totals are deterministic for identical input logs.
- Aggregation does not fail on a single malformed record.
- A clear confidence/source marker is exposed for OpenCode-derived usage.
- New tests cover OpenCode usage paths with meaningful failure cases.

## Suggested implementation areas
- `src/codex_autorunner/agents/opencode/runtime.py`
- `src/codex_autorunner/agents/opencode/run_prompt.py`
- `src/codex_autorunner/integrations/agents/opencode_backend.py`
- `src/codex_autorunner/core/usage.py`
- `src/codex_autorunner/surfaces/web/routes/usage.py`
- `src/codex_autorunner/surfaces/web/hub_routes.py`
- `tests/test_opencode_runtime.py`
- `tests/test_usage.py`

## Priority
High (blocks confidence in keeping/promoting token analytics UX long-term).

### PR DESCRIPTION
**Pull request by Git-on-my-level**
## Summary
- persist canonical OpenCode usage snapshots per turn into CAR-owned state at `.codex-autorunner/usage/opencode_turn_usage.jsonl`
- extend OpenCode runtime/backend flow to surface final normalized usage snapshot and persist it from all OpenCode execution paths (backend turns, review `run_prompt`, and file chat)
- harden usage aggregation:
  - skip malformed Codex timestamps without aborting file aggregation
  - prefer persisted OpenCode usage when present, fallback to `.opencode/sessions` only when missing
  - add deterministic cumulative-vs-delta fallback semantics for OpenCode session payloads
- expose usage source/confidence metadata in usage summaries and include it in `/api/usage` + `/hub/usage`
- add targeted tests for malformed timestamps, fallback semantics, persisted-preferred behavior, canonical usage snapshot return, and backend canonical token usage persistence

## Validation
- `.venv/bin/python -m pytest tests/test_usage.py tests/test_opencode_runtime.py tests/test_opencode_backend_streaming.py tests/test_backend_run_event_contract.py`
- full pre-commit/CI hooks via `git commit`:
  - fast Discord guardrails
  - black/ruff
  - mypy
  - eslint
  - `pnpm run build`
  - full pytest suite (`2305 passed, 3 skipped`)
  - Hermes readiness scorecard

Closes #803

### CODE REVIEW

File: src/codex_autorunner/surfaces/web/routes/file_chat.py [hunk 1]
```diff
@@ -858,6 +859,14 @@ async def _part_handler(
             interrupt_task.cancel()
             await supervisor.mark_turn_finished(repo_root)
 
+        if output_result.usage:
+            persist_opencode_usage_snapshot(
```
[the remark this answers is not in the document]
Comment (Git-on-my-level) [reply src/codex_autorunner/surfaces/web/routes/file_chat.py]: Fixed in 9b74234e. The file-chat fallback path now preserves `output_result.usage` (and keeps an existing `output_result.error`), so usage snapshots are still persisted when streamed text is empty and fallback text comes from `prompt_response`.
